engine/ddf/src/ddfc.py

```diff
@@ -276,7 +276,7 @@ def to_cxx_enum(context, pp, message_type):
 def to_cxx_default_value_string(context, f):
     # Skip all empty values. Type string is an exception as we always set these to ""
     if len(f.default_value) == 0 and f.type != FieldDescriptor.TYPE_STRING:
-        return '0x0'
+        return ''
```

Comment:
If leaving the string empty, it seems to correctly produce a 0 value of the correct length.

Removing _lots_ of compiler warnings.